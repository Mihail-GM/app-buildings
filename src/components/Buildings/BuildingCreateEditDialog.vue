<template>
	<v-dialog
		v-model="dialog"
		max-width="600px"
		@click:outside=close
	>
		<template v-slot:activator="{ on, attrs }">
			<v-btn
				color="primary"
				dark
				class="mb-2 px-8"
				:color="'green'"
				v-bind="attrs"
				v-on="on"
			>
				Add building
			</v-btn>
		</template>

		<v-card>
			<v-card-title class="d-flex justify-center">
				<span id="dialog-title" class="text-h5 text-center">{{ formTitle }}</span>
			</v-card-title>

			<v-card-text>
				<v-container>
					<v-row>
						<v-col
							cols="12"
							sm="12"
							md="12"
							class="py-0"
						>
							<v-text-field
								:disabled="!isCreateTable"
								v-model="editedItem.id"
								dense
								label="Building id"
								outlined
								required
								number
								:error-messages="buildingIdError"
								@blur="$v.editedItem.id.$touch()"
							/>

							<v-text-field
								v-model="editedItem.name"
								dense
								label="Building name"
								outlined
								required
								:error-messages="buildingNameError"
								@blur="$v.editedItem.name.$touch()"
							/>

							<v-text-field
								v-model="editedItem.area"
								dense
								label="Building area (m2)"
								outlined
								required
								type="number"
								:error-messages="buildingAreaError"
								@blur="$v.editedItem.area.$touch()"
							/>

							<v-text-field
								v-model="editedItem.location"
								dense
								label="Building location"
								outlined
							/>

							<v-select
								v-model="editedItem.type"
								dense
								label="Building type"
								:items="types"
								item-text="name"
								item-value="value"
								outlined
							/>
						</v-col>
					</v-row>
				</v-container>
			</v-card-text>

			<v-card-actions>
				<v-spacer></v-spacer>
				<v-btn
					id="cancel-button"
					color="red darken-1"
					style="color:white"
					@click.native="close"
					text
				>
					Cancel
				</v-btn>

				<v-btn
					id="create-edit-button"
					color="green darken-1"
					style="color:white"
					@click.native="save"
					:disabled="$v.$anyError || buildingIdError.length !== 0"
					text
				>
					{{ buttonText }}
				</v-btn>
			</v-card-actions>
		</v-card>
	</v-dialog>
</template>

<script>
	import {validationMixin} from 'vuelidate'
	import {required, minValue} from 'vuelidate/lib/validators'
	import {mapGetters} from "vuex";

	export default {
		name: "BuildingCreateEditDialog",

		components: {},

		props: {
			value: {
				Boolean,
				default: false
			},

			editedItemProp: {
				type: Object,
				required: true,
				default: {
					id: null,
					name: '',
					area: '',
					location: '',
					type: '',
					img: '',
				}
			},

			editedIndexProp: {
				type: Number,
				required: true,
				default: false
			},
		},

		mixins: [validationMixin],

		validations: {
			editedItem: {
				id: {required, minValue: minValue(1)},
				name: {required},
				area: {required, minValue: minValue(1)},
			}
		},

		data() {
			return {
				complete: false,
				editedItem: this.editedItemProp,
				editedIndex: this.editedIndexProp,

				defaultItem: {
					id: null,
					name: '',
					area: null,
					location: '',
					type: null,
					img: '',
				},

				types: [
					{
						value: 'residential',
						name: 'Residential'
					},
					{
						value: 'industrial',
						name: 'Industrial'
					},
					{
						value: 'educational',
						name: 'Educational'
					},
					{
						value: 'special',
						name: 'Special'
					},
				]
			}
		},

		methods: {
			close() {
				this.dialog = false;
				this.$emit('resetEditedItem');
				this.editedIndex = -1;
				this.editedItem = Object.assign({}, this.defaultItem);
				this.$v.$reset();
			},

			save() {
				this.$v.$touch();

				if (this.$v.$anyError) {
					this.$store.dispatch('snackbar/showSnackbarMessage', {
						message: 'Please enter valid data',
						duration: 6000,
						mode: 'fail'
					});
					return
				}

				this.$emit('save', this.editedItem);
				this.close();
			},
		},

		computed: {
			...mapGetters(
				{
					buildings: 'buildings/getBuildings',
				}
			),

			isCreateTable() {
				return this.editedIndex === -1;
			},

			formTitle() {
				return this.isCreateTable ? 'Create new building' : 'Edit building';
			},

			buttonText() {
				return this.isCreateTable ? 'CREATE PROJECT' : 'EDIT PROJECT';
			},

			isIdUniqueRequired() {
				return this.isCreateTable && this.buildings.some(oneBuilding => oneBuilding.id === this.editedItem.id)
			},

			dialog: {
				get() {
					return this.value
				},

				set(value) {
					this.$emit('input', value)
				}
			},

			buildingIdError() {
				const errors = []
				if (!this.$v.editedItem.id.$dirty) {
					return errors
				}

				!this.$v.editedItem.id.minValue && errors.push('Enter positive number for id')
				!this.$v.editedItem.id.required && errors.push('Building id is required.')
				this.isIdUniqueRequired && errors.push('Building id needs to be unique.')

				return errors
			},

			buildingNameError() {
				const errors = []
				if (!this.$v.editedItem.name.$dirty) {
					return errors
				}

				!this.$v.editedItem.name.required && errors.push('Building name is required.')
				return errors
			},

			buildingAreaError() {
				const errors = []
				if (!this.$v.editedItem.area.$dirty) {
					return errors
				}

				!this.$v.editedItem.area.minValue && errors.push('Enter positive number for area')
				!this.$v.editedItem.area.required && errors.push('Building area is required.')
				return errors
			},
		},
	}
</script>

<style lang="scss" scoped>
	.v-label {
		font-size: 14px !important;
	}

	.stripe-card {
		width: 100%;
		height: 45px;
		border-radius: 4px;
	}
</style>