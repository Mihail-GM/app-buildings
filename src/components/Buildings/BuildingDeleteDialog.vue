<template>
	<v-dialog v-model="dialogDelete" max-width="500px">
		<v-card >
			<v-card-title style="background: #000000;">
				<h4 class="white--text"> Delete Building </h4>
			</v-card-title>

			<v-card-subtitle class="py-10 text--primary text-justify">
				This will delete the building, all data for this building. This cannot be undone.
				To confirm you wish to delete all data related to this building, enter the building
				name in the box below, and click CONFIRM.
			</v-card-subtitle>

			<v-card-text>
				<v-text-field
					dense
					v-model="inputName"
					label="Building name"
					outlined
				>
				</v-text-field>
			</v-card-text>

			<v-card-actions>
				<v-spacer></v-spacer>
				<v-btn
					id="cancel-button"
					color="red darken-1"
					text
					@click.native="closeDelete"
				>
					Cancel
				</v-btn>

				<v-btn
					id="confirm-delete-button"
					color="green darken-1"
					text
					@click.native="deleteItemConfirm"
				>
					CONFIRM
				</v-btn>
				<v-spacer></v-spacer>
			</v-card-actions>
		</v-card>
	</v-dialog>
</template>

<script>
	export default {
		name: "BuildingDeleteDialog",

		props: {
			value: {
				Boolean,
				default: false
			},

			editedItemProp: {
				type: Object,
				required: true,
				default: {
					uuid: null,
					id: null,
					name: '',
					area: null,
					location: '',
					type: null,
					img: '',
				}
			},
		},

		data() {
			return {
				editedItem: this.editedItemProp,
				inputName: '',
			}
		},

		methods: {
			deleteItemConfirm() {
				if (this.editedItemProp.name === this.inputName) {
					this.$emit('deleteItemConfirm', this.editedItem);
					this.$store.dispatch('snackbar/showSnackbarMessage', {
						message: 'Building was deleted successfully',
						duration: 4000,
						mode: 'success'
					});
					this.closeDelete();
					return;
				}

				this.$store.dispatch('snackbar/showSnackbarMessage', {
					message: 'To delete building names must mach',
					duration: 6000,
					mode: 'fail'
				});
			},

			closeDelete() {
				this.dialogDelete = false;
			},
		},

		computed: {
			dialogDelete: {
				get() {
					return this.value
				},

				set(value) {
					this.$emit('input', value)
				}
			},
		},
	}
</script>

<style scoped>

</style>