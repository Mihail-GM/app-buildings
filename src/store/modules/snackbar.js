const state = {
	snackbarMessageData: {
		message: '',
		duration: 3000,
		mode: 'success'
	},
};

const getters = {
	getSnackbarMessage(state) {
		return state.snackbarMessageData;
	},
};

const mutations = {
	setSnackbarMessage(state, {message, duration = 4000, mode = 'success', buttonText = 'Close'}) {
		state.snackbarMessageData = {message, duration, mode, buttonText};
	},
}

const actions = {
	showSnackbarMessage(vuexContext, messageData) {
		vuexContext.commit('setSnackbarMessage', messageData);
	},
}

export default {
	namespaced: true,
	state,
	getters,
	mutations,
	actions
}