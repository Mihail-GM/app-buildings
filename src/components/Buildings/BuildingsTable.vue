<template>
	<div id="buildings-table-container" class="mb-16">
		<div id="buildings-table-loading" v-if="actionState === 'LOADING'">
			<v-progress-linear
				indeterminate
				color="green"
			>

			</v-progress-linear>
			<div class="d-flex justify-center align-center pa-10">
				Loading... Please wait
			</div>
		</div>

		<v-alert
			v-else-if="actionState === 'ERRORED'"
			id="buildings-table-errored"
			dense
			outlined
			type="error"
		>
			There was an error when loading data. Please try again later.
		</v-alert>

		<v-data-table
			v-else
			id="buildings-table"
			:headers="headers"
			:items="buildings"
			sort-by="primary_contact_email"
			class="elevation-2 px-10 text-xl-h4"
		>
			<template v-slot:item.img="{ item }">
				<div class="d-flex justify-center align-center py-1">
					<v-img
						v-if="item.type"
						class="pt-5 mt-2"
						max-height="60"
						max-width="60"
						:src="getImgUrl(item.type)"
						:alt="imageData[item.type].alt"
						:title="imageData[item.type].title"
					/>
				</div>
			</template>

			<template v-slot:top>
				<v-toolbar
					flat
				>
					<v-spacer></v-spacer>

					<BuildingCreateEditDialog
						:edited-item-prop.sync="editedItem"
						:key="dialogKeyCreateEditDialog + '-create-edit'"
						:edited-index-prop="editedIndex"
						v-model="dialog"
						@resetEditedItem="resetEditedItem"
						@save="save"
					/>

					<BuildingDeleteDialog
						:key="dialogKeyDeleteDialog + '-delete'"
						v-model="dialogDelete"
						:edited-item-prop.sync="editedItem"
						@deleteItemConfirm="deleteItemConfirm"
					/>
				</v-toolbar>
			</template>

			<template v-slot:item.actions="{ item, index }">
				<v-icon
					:id="'edit-building-button-' + index"
					class="mr-2"
					@click.stop="editItem(item)"
				>
					mdi-pencil
				</v-icon>

				<v-icon
					:id="'delete-building-button-' + index"
					@click.stop="openDeleteItemDialog(item)"
				>
					mdi-delete
				</v-icon>
			</template>
		</v-data-table>
	</div>
</template>

<script>

	import BuildingCreateEditDialog from "./BuildingCreateEditDialog";
	import BuildingDeleteDialog from "./BuildingDeleteDialog";
	import BuildingsService from "@/services/buildings.service";
	import {mapGetters} from "vuex";

	export default {
		name: 'BuildingsTable',

		components: {
			BuildingDeleteDialog,
			BuildingCreateEditDialog
		},

		data() {
			return {
				dialog: false,
				dialogDelete: false,
				dialogKeyCreateEditDialog: 0,
				dialogKeyDeleteDialog: 0,

				headers: [
					{text: 'ID', align: 'start', sortable: true, value: 'id'},
					{text: 'Name', align: 'start', sortable: true, value: 'name'},
					{text: 'Area', align: 'start', sortable: true, value: 'area'},
					{text: 'Location', align: 'start', sortable: true, value: 'location'},
					{text: 'Type', align: 'start', sortable: true, value: `type`},
					{text: 'Image', align: 'center', sortable: false, value: `img`},
					{text: '', value: 'actions', sortable: false, align: 'end'},
				],

				editedIndex: -1,
				defaultItem: {
					uuid: null,
					id: null,
					name: '',
					area: '',
					location: '',
					type: '',
				},
				editedItem: {
					uuid: null,
					id: null,
					name: '',
					area: '',
					location: '',
					type: '',
				},

				imageData: {
					residential: {
						path: 'img/type-apartment.png',
						alt: 'Type - residential',
						title: 'Residential'
					},

					industrial: {
						path: 'img/industrial-building.png',
						alt: 'Type - industrial',
						title: 'Industrial'
					},

					educational: {
						path: 'img/educational-building.png',
						alt: 'Type - educational',
						title: 'Educational'
					},

					special: {
						path: 'img/special-building.png',
						alt: 'Type - special',
						title: 'Special'
					},
				}
			}
		},

		methods: {
			getAllBuildings() {
				this.$store.dispatch('buildings/fetchBuildings');
			},

			editItem(item) {
				this.editedIndex = this.buildings.indexOf(item);
				this.dialogKeyCreateEditDialog++;
				this.editedItem = Object.assign({}, item);
				this.dialog = true;
			},

			openDeleteItemDialog(item) {
				this.editedIndex = this.buildings.indexOf(item);
				this.dialogKeyDeleteDialog++;
				this.editedItem = Object.assign({}, item);
				this.dialogDelete = true;
			},

			resetEditedItem() {
				this.editedIndex = -1;
				this.editedItem = Object.assign({}, this.defaultItem);
			},

			save(editedItemFromForm) {
				if (this.editedIndex > -1) {
					BuildingsService.editBuilding(editedItemFromForm)
						.then((res) => {
							this.$store.dispatch('buildings/fetchBuildings');

							this.$store.dispatch('snackbar/showSnackbarMessage', {
								message: 'Building was edited successfully',
								duration: 4000,
								mode: 'success'
							});
						});

					return;
				}

				BuildingsService.createBuilding(JSON.stringify(editedItemFromForm))
					.then((res) => {
						this.$store.dispatch('buildings/fetchBuildings');

						this.$store.dispatch('snackbar/showSnackbarMessage', {
							message: 'Building was added successfully',
							duration: 4000,
							mode: 'success'
						});

						return res;
					});
			},

			closeDelete() {
				this.dialogDelete = false;
				this.$nextTick(() => {
					this.resetEditedItem();
				});
			},

			deleteItemConfirm(deletedItem) {
				BuildingsService.deleteBuilding(deletedItem)
					.then((res) => {
						this.$store.dispatch('buildings/fetchBuildings');

						this.$store.dispatch('snackbar/showSnackbarMessage', {
							message: 'Building was deleted successfully',
							duration: 4000,
							mode: 'success'
						});

						return res;
					});

				this.closeDelete();
			},

			getImgUrl(type) {
				return require('@/assets/' + (this.imageData[type].path))
			}
		},

		computed: {
			...mapGetters(
				{
					buildings: 'buildings/getBuildings',
					actionState: 'buildings/getCurrentState'
				}
			)
		},
		created() {
			this.getAllBuildings();
		},
	}
</script>
