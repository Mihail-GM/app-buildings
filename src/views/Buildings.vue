<template>
	<div>
		<Header/>

		<TheLayoutWrap>
			<BuildingsTable/>
		</TheLayoutWrap>
	</div>
</template>

<script>
	import BuildingsTable from "../components/Buildings/BuildingsTable";
	import Header from "../components/Base/Header";
	import TheLayoutWrap from "../layouts/TheLayoutWrap";

	export default {
		name: "Buildings",
		components: {TheLayoutWrap, Header, BuildingsTable},
	}
</script>

<style scoped>

</style>