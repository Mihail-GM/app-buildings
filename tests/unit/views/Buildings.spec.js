// test/Buildings.spec.js

import Vuetify from 'vuetify'
import Vuex from "vuex";

import {createLocalVue, mount} from '@vue/test-utils'
import Buildings from "@/views/Buildings";
import {merge} from "lodash";
import snackbar from "@/store/modules/snackbar";

describe('Buildings.vue', () => {
	const localVue = createLocalVue();
	localVue.use(Vuex);
	let vuetify

	beforeEach(() => {
		vuetify = new Vuetify()
	})

	function createStore(overrides) {
		let snackbarMessageData = {
			message: 'Test',
			duration: 3000,
			mode: 'success'
		};

		const defaultStoreConfig = {
			modules: {
				snackbar: {
					namespaced: true,
					state: {
						snackbarMessageData
					},
					getters: snackbar.getters,
					mutations: snackbar.mutations,
					actions: snackbar.actions,
				}
			}
		};

		return new Vuex.Store(merge(defaultStoreConfig, overrides));
	}

	const createWrapper = (overrides) => {
		const defaultMountingOptions = {
			localVue,
			vuetify,
			store: createStore(),
		};

		return mount(Buildings, merge(defaultMountingOptions, overrides));
	}

	it('should render component Buildings', () => {
		const wrapper = createWrapper()

		expect(wrapper.is(Buildings)).toBe(true);
	})

	it('should match snapshot', () => {
		const wrapper = createWrapper()

		expect(wrapper.html()).toMatchSnapshot()
	})
})
