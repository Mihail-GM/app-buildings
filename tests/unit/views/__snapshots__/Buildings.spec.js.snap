// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buildings.vue should match snapshot 1`] = `
<div>
  <header class="px-md-10 px-sm-0 px-xs-0 v-sheet theme--light v-toolbar v-toolbar--dense v-app-bar v-app-bar--fixed" style="height: 48px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 48px;">
      <div class="v-toolbar__title">
        Welcome
      </div>
      <div class="spacer"></div> <i aria-hidden="true" class="v-icon notranslate mdi mdi-account-circle theme--light" style="font-size: 40px;"></i>
    </div>
  </header>
  <div>
    <main class="v-main mt-5" style="padding: 48px 0px 0px 0px;">
      <div class="v-main__wrap">
        <div class="container">
          <div id="buildings-table-container" class="mb-16">
            <div class="v-data-table elevation-2 px-10 text-xl-h4 v-data-table--has-top v-data-table--has-bottom theme--light" id="buildings-table">
              <header class="v-sheet theme--light v-toolbar v-toolbar--flat" style="height: 64px;">
                <div class="v-toolbar__content" style="height: 64px;">
                  <div class="spacer"></div>
                  <div class="v-dialog__container"><button type="button" class="mb-2 px-8 v-btn v-btn--is-elevated v-btn--has-bg theme--dark v-size--default green" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content">
			Add building
		</span></button>
                    <!---->
                  </div>
                  <div class="v-dialog__container">
                    <!---->
                  </div>
                </div>
              </header>
              <div class="v-data-table__wrapper">
                <table>
                  <colgroup>
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                  </colgroup>
                  <thead class="v-data-table-header">
                    <tr>
                      <th role="columnheader" scope="col" aria-label="ID: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>ID</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Name: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Name</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Area: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Area</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Location: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Location</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Type: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Type</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Image" class="text-center"><span>Image</span></th>
                      <th role="columnheader" scope="col" aria-label="" class="text-end"><span></span></th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr class="v-data-table__empty-wrapper">
                      <td colspan="7">No data available</td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div class="v-data-footer">
                <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
                    <div class="v-input__control">
                      <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-84" class="v-input__slot">
                        <div class="v-select__slot">
                          <div class="v-select__selections">
                            <div class="v-select__selection v-select__selection--comma">10</div><input aria-label="Rows per page:" id="input-84" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                          </div>
                          <div class="v-input__append-inner">
                            <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                          </div><input type="hidden" value="10">
                        </div>
                        <div class="v-menu">
                          <!---->
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="v-data-footer__pagination">â€“</div>
                <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
                <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <nav class="v-navigation-drawer v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light" style="height: 100vh; top: 0px; transform: translateX(-100%); width: 256px;">
      <div class="v-navigation-drawer__content">
        <div class="v-list v-sheet theme--light">
          <router-link tabindex="0" id="mobile-navigation-Home" class="v-list-item v-list-item--link theme--light">
            <div class="v-list-item__action"><i aria-hidden="true" class="v-icon notranslate mdi mdi-home-circle theme--light"></i></div>
            <div class="v-list-item__content">Home</div>
          </router-link>
          <router-link tabindex="0" id="mobile-navigation-Buildings" class="v-list-item v-list-item--link theme--light">
            <div class="v-list-item__action"><i aria-hidden="true" class="v-icon notranslate mdi mdi-office-building theme--light"></i></div>
            <div class="v-list-item__content">Buildings</div>
          </router-link>
        </div>
      </div>
      <div class="v-navigation-drawer__border"></div>
    </nav>
    <header class="px-md-10 px-sm-0 px-xs-0 v-sheet theme--dark v-toolbar v-toolbar--bottom" style="width: 100%; height: 64px; position: fixed;">
      <div class="v-toolbar__content" style="height: 64px;"><span class="hidden-md-and-up"><button type="button" class="v-app-bar__nav-icon v-btn v-btn--icon v-btn--round theme--dark v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu theme--dark"></i></span></button></span> <button type="button" class="hidden-sm-and-down v-btn v-btn--text theme--dark v-size--x-large" id="navigation-home"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-home-circle theme--dark"></i>
			Home
		</span></button><button type="button" class="hidden-sm-and-down v-btn v-btn--text theme--dark v-size--x-large" id="navigation-buildings"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-office-building theme--dark"></i>
			Buildings
		</span></button>
        <div class="spacer"></div>
        <div class="v-toolbar__title">
          <h3 id="app-title-navigation" style="cursor: pointer;">
            Buildings App
          </h3>
        </div>
      </div>
    </header>
  </div>
</div>
`;
